Assets/Scripts/Calendar UI.cs

```diff
@@ -23,89 +25,136 @@ public class CalendarUI : MonoBehaviour
+        numberOfBlanksBefore = GetBlanksBefore();
+        numberOfBlanksAfter = GetBlanksAfter();
+
         PopulateCalendarGrid();
+
     }
+    private void HandleBackwardClick()
+    {
+        calendarGridPopulator.ClearCalendarGrid();
+        defaultDate = defaultDate.AddMonths(-1);
 
 
+        monthText.text = defaultDate.ToString("MMMM");
+        yearText.text = defaultDate.ToString("yyyy");
+        numberOfBlanksBefore = GetBlanksBefore();
+        numberOfBlanksAfter = GetBlanksAfter();
+        PopulateCalendarGrid();
+    }
 
     public void PopulateCalendarGrid()
     {
 
         int totalDays = numberOfBlanksBefore + daysInMonth + numberOfBlanksAfter;
-        List<GameObject> populationList = GridPopulator.instance.PopulateTheGrid(totalDays, calendarGrid, dayPrefab);
+        List<Cell> populationList = calendarGridPopulator.PopulateTheGrid(totalDays);
+        int day = 1;
         for (int i = 0; i <= populationList.Count - 1; i++)
         {
 
 
             if (i < numberOfBlanksBefore || i >= numberOfBlanksBefore + daysInMonth)
             {
-                populationList[i].GetComponent<Image>().color = blankColor;
-                GridPopulator.instance.SetTextValue("", populationList[i]);
+                populationList[i].SetImageColor(blankColor);
+                populationList[i].SetTextValue("");
 
             }
 
             else
             {
-                GridPopulator.instance.SetTextValue("i.ToString()", populationList[i]);
-                if (i == currentDate.Day)
+
+                populationList[i].SetTextValue(day.ToString());
+                if (day == currentDate.Day && defaultDate.Month == currentDate.Month && defaultDate.Year == currentDate.Year)
                 {
-                    populationList[i].GetComponent<Image>().color = currentDayColor;
+                    populationList[i].SetImageColor(currentDayColor);
                 }
                 else
                 {
-                    populationList[i].GetComponent<Image>().color = daysDefaultColor;
+                    populationList[i].SetImageColor(daysDefaultColor);
                 }
-
+                day++;
             }
-            populationList.Add(dayPrefab);
+
         }
 
 
     }
 
+
+
     private int GetBlanksBefore()
     {
+
+        firstDayOfMonth = new DateTime(defaultDate.Year, defaultDate.Month, 1);
         int dayofWeek = (Int32)firstDayOfMonth.DayOfWeek;
-        int number = (dayofWeek + 6) % 7;
+        int number = (dayofWeek + 6) % DAYS_IN_WEEK;
 
         return number;
     }
     private int GetBlanksAfter()
     {
-        DateTime lastDayOfMonth = new DateTime(currentDate.Year, currentDate.Month, daysInMonth);
+        daysInMonth = DateTime.DaysInMonth(defaultDate.Year, defaultDate.Month);
+        DateTime lastDayOfMonth = new(defaultDate.Year, defaultDate.Month, daysInMonth);
         int dayofWeek = (Int32)lastDayOfMonth.DayOfWeek;
```

Comment:
I think we can just cast to (int) instead of fully qualifying an (Int32)

Reply:
Resolved